src/python/pants/option/global_options.py

```diff
@@ -358,20 +380,21 @@ def _use_auth_plugin(
         store_rpc_concurrency = cast(int, bootstrap_options.remote_store_rpc_concurrency)
         cache_rpc_concurrency = cast(int, bootstrap_options.remote_cache_rpc_concurrency)
         execution_rpc_concurrency = cast(int, bootstrap_options.remote_execution_rpc_concurrency)
-        auth_plugin_path, _, auth_plugin_func = bootstrap_options.remote_auth_plugin.partition(":")
-        auth_plugin_module = importlib.import_module(auth_plugin_path)
-        auth_plugin_func = getattr(auth_plugin_module, auth_plugin_func)
         auth_plugin_result = cast(
             AuthPluginResult,
-            auth_plugin_func(
+            remote_auth_plugin_func(
                 initial_execution_headers=execution_headers,
                 initial_store_headers=store_headers,
                 options=full_options,
                 env=dict(env),
                 prior_result=prior_result,
             ),
         )
-        plugin_name = auth_plugin_result.plugin_name or bootstrap_options.remote_auth_plugin
+        plugin_name = (
+            auth_plugin_result.plugin_name
+            or bootstrap_options.remote_auth_plugin
+            or remote_auth_plugin_func.__name__
```

Comment:
Won't `__name__` always be `remote_auth`, since it is required to be for loading? You'll probably want to use the module here instead.

Reply:
updated